src/structs/probability_tables.rs

```diff
@@ -92,17 +91,16 @@ impl ProbabilityTables {
 
     pub fn calc_non_zero_counts_context_7x7<const ALL_PRESENT: bool>(
```

Comment:
I think: what if rename this to `calc_non_zero_counts_predictor_7x7` or smth like that and return `num_non_zeros_above << 1`,  `num_non_zeros_left << 1` or `num_non_zeros_above + num_non_zeros_leftnum_non_zeros_above + num_non_zeros_left. Then by extending array `NON_ZERO_TO_BIN` to 100 bytes we can reduce number of operations per coefficient.

Reply:
I'll separate this in another change... better to separate refactoring from functional changes